uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-request-popup-details-tab.component.spec.js

```diff
@@ -1288,6 +1288,84 @@ define([
       });
     });
 
+    describe('time and date inputs watchers', function () {
+      beforeEach(function () {
+        var absenceTypes = absenceTypeData.all().values;
+
+        compileComponent({
+          mode: 'create',
+          role: 'admin',
+          selectedAbsenceType: absenceTypes[0]
+        });
+        spyOn(controller, 'calculateBalanceChange').and.callThrough();
+      });
+
+      describe('when the calculation unit is "hours"', function () {
+        beforeEach(function () {
+          selectedAbsenceType.calculation_unit_name = 'hours';
+        });
+
+        describe('when from/to deductions values are set but not changed', function () {
+          beforeEach(function () {
+            // the amounts are 0, setting them to 0 again still fires the watcher
+            controller.uiOptions.times.from.amount = 0;
+            controller.uiOptions.times.to.amount = 0;
+
+            $rootScope.$digest();
+          });
+
+          it('does not call the balance change calculation', function () {
+            expect(controller.calculateBalanceChange).not.toHaveBeenCalled();
+          });
+
+          describe('when from/to deductions values are changed', function () {
+            beforeEach(function () {
+              controller.uiOptions.times.from.amount = 80;
+              controller.uiOptions.times.to.amount = 100;
+
+              $rootScope.$digest();
+            });
+
+            it('calls the balance change calculation', function () {
+              expect(controller.calculateBalanceChange).toHaveBeenCalled();
+            });
+          });
+        });
+      });
+
+      describe('when the calculation unit is "days"', function () {
+        beforeEach(function () {
+          selectedAbsenceType.calculation_unit_name = 'days';
+        });
+
+        describe('when from/to deductions are set', function () {
+          beforeEach(function () {
+            controller.uiOptions.times.from.amount = 50;
+            controller.uiOptions.times.to.amount = 100;
+
+            $rootScope.$digest();
+          });
+
+          it('does not call the balance change calculation', function () {
+            expect(LeaveRequestAPI.calculateBalanceChange).not.toHaveBeenCalled();
+          });
+        });
+      });
+
+      describe('when from/to dates are changed', function () {
+        beforeEach(function () {
+          controller.uiOptions.fromDate = '1989-04-14';
+
+          spyOn(controller.request, 'getWorkDayForDate').and.returnValue($q.reject());
+          $rootScope.$digest();
+        });
+
+        it('calls the time and deduction ranges loading', function () {
```

Comment:
Can it be something like `loads the time and deduction ranges` or `calls the function to load time and deduction ranges`